user/jwtValidator.go

```diff
@@ -0,0 +1,152 @@
+package user
+
+import (
+	"errors"
+	"log"
+	"net/http"
+	"strings"
+
+	auth0 "github.com/auth0-community/go-auth0"
+	jwt "github.com/dgrijalva/jwt-go"
+	jose "gopkg.in/square/go-jose.v2"
+)
+
+type JWTValidator interface {
+	ValidateRequest(request *http.Request) (*TokenData, error)
+}
+
+type jwtValidator struct {
+	config JWTValidatorConfig
+}
+
+type JWTValidatorConfig struct {
+	Auth0Domain string
+	Secret      string
+}
+
+func NewJWTValidator(config JWTValidatorConfig) (*jwtValidator, error) {
+	if config.Auth0Domain == "" {
+		return nil, errors.New("The Auth0 domain needs to be set.")
+	}
+	if config.Secret == "" {
+		return nil, errors.New("The signing secret needs to be set.")
+	}
+	return &jwtValidator{config: config}, nil
+}
+
+func getAccessToken(request *http.Request) string {
+	if request != nil {
+		auth := request.Header.Get("Authorization")
+		if len(auth) > 7 &&
+			strings.EqualFold(auth[0:7], "BEARER ") {
+			return strings.Split(auth, " ")[1]
+		}
+	}
+	return ""
+}
+
+func isSessionToken(request *http.Request) bool {
+	return getSessionToken(request) != ""
+}
+
+func isAccessToken(request *http.Request) bool {
+	return getAccessToken(request) != ""
+}
+
+func getSessionToken(request *http.Request) string {
+	if request != nil {
+		return request.Header.Get(TP_SESSION_TOKEN)
+	}
+	return ""
+}
+
+func makeAccessRequest(token string) *http.Request {
```

Comment:
What are `makeAccessRequest` and `makeSessionRequest` used for? They seem very specific to the `/` path. Plus the error is ignored.